grails-app/services/org/broadinstitute/orsp/DataUseLetterService.groovy

```
@@ -56,5 +58,13 @@ class DataUseLetterService {
         restriction
     }
 
+    DataUseRestriction updateRestrictionFromParams(Object params) {
+        if (params.consentGroupKey) {
+            DataUseRestriction existentRestriction = DataUseRestriction.findByConsentGroupKey(params.consentGroupKey)
```

Should probably throw the not found exception here if not found.

---

If it is not found, it will be created in DataUseRestrictionParser

---

Gotcha, yes, thank you!